### ISSUE TITLE
Original and destination links are merged when copying to the clipboard

### ISSUE THREAD

**Issue by birdie-github** [Cimbali/CleanLinks#43]
In the log window I see

    $Source_URL
    → $Cleaned_URL

However when I copy it to the clipboard, I get

    $Source_URL$Cleaned_URL

which is not exactly readable.

If you're OK with that then close this bug report as invalid.

**Comment by Cimbali** [comment]
I have a newline between `$source_url` and `$cleaned_url`. If you do not have it, please report your firefox version, platform, etc. For example, maybe on Windows `\n\r` should be the separator, instead of just `\n`.

If you have any suggestion for a better separator than a new line between both URLs, do not hesitate.

**Comment by birdie-github** [comment]
This is 100% reproducible under Linux / Firefox 63 beta 9 ( https://ftp.mozilla.org/pub/firefox/releases/63.0b9/linux-x86_64/en-US/ ).

**Comment by Cimbali** [comment]
So missing newline on Linux 64 bits with ff 63 beta 9.

**Comment by birdie-github** [comment]
Yep.

I wonder if using " → " (space right arrow space) is sufficient. It's not such a common symbol for URLs - in fact I'm not sure it can be used as is.

**Comment by birdie-github** [comment]
Nope, I'm running on real hardware.

**Comment by Cimbali** [comment]
I have the newline on Firefox 72.0. Will to test other environments.

**Comment by Cimbali** [comment]
Also works on FF 68 and 72 on Linux desktops, FF 64 on Windows, FF 68 on Android. Closing until we can reproduce somewhere.

**Comment by birdie-github** [comment]
I can reproduce it in Firefox 74 under Fedora 31/XFCE right now.

`https://www.imdb.com/title/tt2527338/?ref_=rt_li_tthttps://www.imdb.com/title/tt2527338/`

**Comment by Cimbali** [comment]
How are you copying?

- Are you actually selecting text or just clicking an item in the cleaned list and doing ctrl + c? (Both cases work for me, but you don’t need to select it.)
- Where are you pasting this into?
- Is something else interfering with your clipboard?

**Comment by birdie-github** [comment]
1. Click the extension icon
2. Click the log item I'm interested in to expand it
3. Use mouse to make a selection
4. Ctrl + C

5. Ctrl + V here

Just for fun I've just clicked Ctrl + A and now I'm doing Ctrl + V:
```

Clean Links v4.1.2 — Enabled

Filter link type
Filter cleaning action

https://www.imdb.com/title/tt2527338/?ref_=rt_li_tthttps://www.imdb.com/title/tt2527338/


```

I've got no apps interfering with the buffer.

**Comment by Cimbali** [comment]
Alright, can you try your steps without step 3? So:

1. Click the extension icon
2. Click the log item you're interested in to expand it
3. Ctrl + C
4. Ctrl + V here

**Comment by birdie-github** [comment]
Ctrl+C does nothing in this case. Nothing gets copied to the clipboard.

**Comment by Cimbali** [comment]
Ok, that’s really how it should work though. So the 'copy' event does not get triggered for some reason

https://github.com/Cimbali/CleanLinks/blob/d57cde1f41a3d10c3444c0bb11154b756d3a01a0/addon/pages/popup.js#L238-L248

**Comment by Cimbali** [comment]
So does the live example [on this MDN page](https://developer.mozilla.org/en-US/docs/Web/API/Element/copy_event) work for you?

**Comment by birdie-github** [comment]
If I understand correctly pressing Ctrl + C while this page is open should copy "Try copying text from this box..." to the clipboard, right?

This doesn't happen. I even placed the cursor in the middle of the text - no dice.

**Comment by Cimbali** [comment]
No you have to select some text in “Try copying text from this box” box. When you copy it, your clipboard should actually contain the selected text but in upper case.

**Comment by birdie-github** [comment]
Nope, it's copied as as without case conversion.

**Comment by Cimbali** [comment]
Does the “copy clean links” work, from the context menu? That’s using another way of interacting with the clipboard.

**Comment by birdie-github** [comment]
> Does the “copy clean links” work, from the context menu? That’s using another way of interacting with the clipboard.

Yes, it works perfectly.

**Comment by Cimbali** [comment]
Good, then I’ll make a copy button in the popup with that functionality. That will also be useful for android interfaces.

**Comment by Cimbali** [comment]
Closed by 1f8eccd (with wrong issue number in message…)
